File: HackerRank/Python/Easy/E0007.py
# Problem Statement: https://www.hackerrank.com/challenges/list-comprehensions/problem

X, Y, Z, N = (int(input()) for _ in range(4))
l = [[i, j, k] for i in range(X + 1) for j in range(Y + 1) for k in range(Z + 1) if i + j + k != N]
print(l)

File: HackerRank/C++/Easy/E0007.cpp
/*
Problem Statement: https://www.hackerrank.com/challenges/c-tutorial-pointer/problem
*/

#include <iostream>
#include <cmath>

using namespace std;

void update(int *a,int *b) {
	int temp = *a + *b;
	*b = abs(*a - *b);
	*a = temp;
}

int main() {
	int a, b;
	cin >> a >> b;
	update(&a, &b);
	cout << a << endl;
	cout << b << endl;
	return 0;
}